numpy/_core/src/umath/ufunc_object.c

```diff
@@ -4440,47 +4186,68 @@ replace_with_wrapped_result_and_return(PyUFuncObject *ufunc,
         ufunc_full_args full_args, npy_bool subok,
         PyArrayObject *result_arrays[])
 {
-    PyObject *retobj[NPY_MAXARGS];
-    PyObject *wraparr[NPY_MAXARGS];
-    _find_array_wrap(full_args, subok, wraparr, ufunc->nin, ufunc->nout);
+    PyObject *result = NULL;
+    PyObject *wrap, *wrap_type;
 
-    /* wrap outputs */
-    for (int i = 0; i < ufunc->nout; i++) {
-        _ufunc_context context;
+    if (!subok) {
+        /* subok=False ignores input wrapping (but not output) */
+        Py_INCREF(Py_None);
+        wrap = Py_None;
+        Py_INCREF(&PyArray_Type);
+        wrap_type = (PyObject *)&PyArray_Type;
+    }
+    else if (npy_find_array_wrap(
+            ufunc->nin, PySequence_Fast_ITEMS(full_args.in),
+            &wrap, &wrap_type) < 0) {
+        goto fail;
+    }
 
-        context.ufunc = ufunc;
-        context.args = full_args;
-        context.out_i = i;
+    /* wrap outputs */
+    NpyUFuncContext context = {
+            .ufunc = (PyObject *)ufunc,
+            .in = full_args.in, .out = full_args.out};
 
-        retobj[i] = _apply_array_wrap(wraparr[i], result_arrays[i], &context);
-        result_arrays[i] = NULL;  /* Was DECREF'ed and (probably) wrapped */
-        if (retobj[i] == NULL) {
+    if (ufunc->nout != 1) {
+        result = PyTuple_New(ufunc->nout);
+        if (result == NULL) {
             goto fail;
         }
     }
 
-    if (ufunc->nout == 1) {
-        return retobj[0];
-    }
-    else {
-        PyObject *result = PyTuple_New(ufunc->nout);
-        if (result == NULL) {
-            return NULL;
+    for (int out_i = 0; out_i < ufunc->nout; out_i++) {
+        context.out_i = out_i;
+        PyObject *original_out = NULL;
+        if (full_args.out) {
+            original_out = PyTuple_GET_ITEM(full_args.out, out_i);
+        }
+
+        PyObject *ret_i = npy_apply_wrap(
+                (PyObject *)result_arrays[out_i], original_out, wrap, wrap_type,
+                /* Always try to return a scalar right now: */
+                &context, PyArray_NDIM(result_arrays[out_i]) == 0, NPY_TRUE);
```

Comment:
Here, the forcing is consistent with the previous behaviour. I think `reduce` always behaved differently...